### ISSUE TITLE
Releases link against debug and release runtime on MSVC

### ISSUE THREAD

**Issue by selaux** [ja2-stracciatella/ja2-stracciatella#641]
Basically we need to redistribute debug and release dlls with MSVC releases. I am not sure why exactly.

I dont consider this a critical as it does not increase release size that much, it just clutters the directory.

**Comment by crusher28** [comment]
I get a msvc error in build 0.16. I test msvc64 + msvc32. Allways the same problem. I have win 7, 64bit.
Microsoft visual c++ 2015 (x64) + (x86) is installed. How can i solve the problem?

**Comment by selaux** [comment]
@crusher28 Can you provide more information (the exact error for example). And what exact build are you using? I did not update the builds attached to the GitHub release yet, so the most current ones should be from here: http://builds-ja2.stefanlau.com/releases/

**Comment by crusher28** [comment]
thx for your answer.  this builds work now: http://builds-ja2.stefanlau.com/releases/

I download first this: https://github.com/ja2-stracciatella/ja2-stracciatella/releases
The error message was: msvcp140.dll not found.
But in the new build, i see there is the msvcp140.dll and the msvcp140d.dll in the stracciatella folder.
All good now.

thx for your help!

I have a last question. Is reslolution 2560 X 1440 planned for the launcher?

**Comment by selaux** [comment]
It is not planned as one to be selected through the dropdown. You can enter it manually as custom resolution though. Stuff will be tiny though 😉

**Comment by selaux** [comment]
This is not fixed yet. Release builds still link against the debug runtime for MSVC. Even more weird: This seems to change during build (uses `\MD` (release) first and then it switches to`\MDd` (debug) :confused:)

**Comment by flaviojs** [comment]
I would consider this as critical! Unless you know what you are doing, there is a real danger of memory corruption while using both at the same time.

Memory corruption is very hard to detect and trace back. I'm pretty sure the ja2 source already corrupts memory by itself under certain (unknown) conditions, it's better not to have another possible cause.

I propose adding mingw releases and discouraging using msvc ones until this is fixed. v0.16.x does not have mingw releases, v0.15.1 is already mingw, older ones are probably mingw (haven't checked).

**Comment by flaviojs** [comment]
Now that I have access to a windows build environment, I decided to give this issue a try...

According to https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=vs-2019#what-problems-exist-if-an-application-uses-more-than-one-crt-version :

> It's also possible to avoid some of these issues if all of the images in your process use the same dynamically loaded version of the CRT. To ensure that all components use the same DLL version of the CRT, build them by using the /MD option, and use the same compiler toolset and property settings.

In the SDL source the project in the VisualC dir always uses `\MD`, even on debug builds.
All dlls in `dependencies/lib-SDL2-2.0.8-VC` link against `MSVCRT.dll`.

This means we should make sure that everything else compiles with `\MD` or links against a release MSVCRT dll.

**Comment by flaviojs** [comment]
Rust uses `/MT` with `crt-static`, and `/MD` otherwise:
 + https://doc.rust-lang.org/1.32.0/reference/linkage.html#static-and-dynamic-c-runtimes
 + https://github.com/rust-lang/rfcs/blob/master/text/1721-crt-static.md

Apparently SDL can be linked with `/MT` code because the VisualC project compiles with `/Zl`:
 + https://stackoverflow.com/a/40905649

**Comment by flaviojs** [comment]
Found the problem. External projects are being compiled as debug (default?) on release builds because CMAKE_BUILD_TYPE does not work as was assumed with multi-configuration projects (VS and XCode):

 + https://stackoverflow.com/a/43363395
 + https://stackoverflow.com/a/30011890

Also, I tried setting the runtime and found that we are not propagating flags to the external projects.

I will try to make everything use `/MD` in MSVC builds.
